File: VB/WindowsApplication2/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Docking

Namespace WindowsApplication2
    Partial Public Class Form1
        Inherits Form

        Private helper As DockHelper

        Public Sub New()
            InitializeComponent()
            helper = New DockHelper(dockManager1)
        End Sub

        Private Sub checkEdit1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkEdit1.CheckedChanged
            Dim check As DevExpress.XtraEditors.CheckEdit = TryCast(sender, DevExpress.XtraEditors.CheckEdit)
            If check IsNot Nothing Then
                helper.AllowDockToTabbedContainer = check.Checked
            End If
        End Sub
    End Class
End Namespace

File: VB/WindowsApplication2/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking;

namespace WindowsApplication2
{
    public partial class Form1 : Form
    {
        DockHelper helper;

        public Form1()
        {
            InitializeComponent();
            helper = new DockHelper(dockManager1);
            helper.AllowDockToSplitContainer = true;
        }


        protected override void OnFormClosing(FormClosingEventArgs e)
        {
            base.OnFormClosing(e);
            helper.AllowDockToSplitContainer = false;
        }
    }
}